extensions/arc/src/ui/dialogs/connectSqlDialog.ts

```diff
@@ -112,19 +112,24 @@ export class ConnectToSqlDialog extends InitializingComponent {
 			connectionProfile.id = result.connectionId;
 			const credentialProvider = await azdata.credentials.getProvider(credentialNamespace);
 			if (connectionProfile.savePassword) {
-				await credentialProvider.saveCredential(createCredentialId(this._controllerModel.info.id, this._miaaModel.info.resourceType, this._miaaModel.info.name), connectionProfile.password);
+				await credentialProvider.saveCredential(createCredentialId(this._controllerModel.info.id, this._model.info.resourceType, this._model.info.name), connectionProfile.password);
 			} else {
-				await credentialProvider.deleteCredential(createCredentialId(this._controllerModel.info.id, this._miaaModel.info.resourceType, this._miaaModel.info.name));
+				await credentialProvider.deleteCredential(createCredentialId(this._controllerModel.info.id, this._model.info.resourceType, this._model.info.name));
 			}
 			this._completionPromise.resolve(connectionProfile);
 			return true;
 		}
 		else {
-			vscode.window.showErrorMessage(loc.connectToSqlFailed(this.serverNameInputBox.value, result.errorMessage));
+			vscode.window.showErrorMessage(this.connectionFailedMessage(result.errorMessage));
+			this._completionPromise.reject(result.errorMessage);
```

Comment:
Oh wait that's right, this is in the validate function which is used to control whether the dialog is closed or not. So we shouldn't actually reject there since returning false means the dialog will stay open. Closing the dialog through other means such as cancel is going to resolve as expected.

So yeah sorry about that but you can just remove the reject line here.